patches/api/0326-Expose-Player-Idle-Time.patch

```diff
@@ -0,0 +1,25 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: thamid-gamer <60953955+thamid-gamer@users.noreply.github.com>
+Date: Sat, 31 Jul 2021 03:42:06 -0400
+Subject: [PATCH] Expose Player Idle Time
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 5e8814cc317a705eaf8bdd9f3876a5366c0a0226..f8d64307dd3bea8f416aae94a4c53ec5f6e467ce 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -2011,6 +2011,14 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+      */
+     void resetCooldown();
+ 
++    /**
++     * Gets the time the player has been idle in milliseconds.
++     * Any packets sent from this player will reset the idle time.
```

Comment:
I would suggest changing this to something along the lines of "most packets caused by player interaction will reset the idle time".

Reply:
Specifically Click Window, Enchant Item, Update Sign, Player Digging, Player Block Placement, Held Item Change, Animation (swing arm), Entity Action, Client Status, Chat Message, Use Entity

Reply:
packets change over time, I will change to most packets

Reply:
I would also mention that this relates to the `server.properties`'s `player-idle-timeout` setting so people can make another connection.